detox/detox.d.ts

```diff
@@ -1279,12 +1279,31 @@ declare global {
              * Performs the action repeatedly on the element until an expectation is met
              * @example await waitFor(element(by.text('Item #5'))).toBeVisible().whileElement(by.id('itemsList')).scroll(50, 'down');
              */
-            whileElement(by: NativeMatcher): NativeElement & WaitFor;
+            whileElement(by: NativeMatcher): NativeElementWaitableActions & WaitFor;
 
             // TODO: not sure about & WaitFor - check if we can chain whileElement multiple times
         }
 
-        interface NativeElementActions {
+        interface NativeElementWaitableActions {
+
+          /**
+           * Scrolls a given amount of pixels in the provided direction, starting from the provided start positions.
+           * @param pixels - independent device pixels
+           * @param direction - left/right/up/down
+           * @param startPositionX - the X starting scroll position, in percentage; valid input: `[0.0, 1.0]`, `NaN`; default: `NaN`—choose the best value automatically
+           * @param startPositionY - the Y starting scroll position, in percentage; valid input: `[0.0, 1.0]`, `NaN`; default: `NaN`—choose the best value automatically
+           * @example await element(by.id('scrollView')).scroll(100, 'down', NaN, 0.85);
+           * @example await element(by.id('scrollView')).scroll(100, 'up');
+           */
+          scroll(
```

Comment:
Please add an extra test to:

https://github.com/wix/Detox/blob/feat/refine_whileElement_api/detox/test/types/detox-global-tests.ts#L93

Invoke the `waitFor(...).scroll(pixels, direction, startX, startY`, in addition to `(pixels, directions)`.

Besides, you can add one more test like `// @ts-expect-error` to ensure that `waitFor(...)` does not have `tap` or some other common method.